external/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -51,11 +52,21 @@ object SchemaConverters {
     toSqlTypeHelper(avroSchema, Set.empty)
   }
 
+  // The property specifies Catalyst type of the given field
+  private val CATALYST_TYPE_PROP_NAME = "spark.sql.catalyst.type"
```

Comment:
Shall we make it `sparkType`?

Reply:
Avro has logical annotation `logicalType`: https://avro.apache.org/docs/1.10.2/spec.html#Logical%20Types
We can have our `sparkType` annotation.

Reply:
> Shall we make it sparkType?

We have already had similar property in ORC, see https://github.com/apache/spark/pull/34184/files#diff-e14fd8725cf71eee7b34fa299c2f3abe5a0033f9abce9de4c7e081ba57991b0bR53 . I would prefer to have the same here for consistency.

> We can have our sparkType annotation.

Do you mean a property? At least in docs of Avro API lib, it is called property, see https://avro.apache.org/docs/current/api/java/org/apache/avro/JsonProperties.html#addProp-java.lang.String-java.lang.String- . That's what I did, isn't it?